### PR TITLE
Improvements

### LINKED ISSUE #3: [Question] Convert To TS
Curious what people would think on converting the code to typescript, and just bundling it into a JS file, and to allow it to still be used on a client, commit that JS file too? Maybe minify it or something

### LINKED ISSUE #4: Add documentation
* Quickstart
* How to use
* methods available

### PR DESCRIPTION
**Pull request by ebebbington**
Fixes #3 and #4

## Summary

* Created a ts file, which is also compiled into a reusable .js file
* Can provide typings now (eg i think of using this in deno)
* Added a readme, documentation and workflows
* Rename to wocket-client

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2019-2020 the sockets authors
+Copyright (c) 2020 the Sockets-client authors
```
Comment (saragee3) [inline LICENSE]: wockets? lol

File: README.md [hunk 2]
```diff
@@ -4,12 +4,85 @@
...
+  <a href="https://github.com/drashland/sockets-client/actions">
+    <img src="https://img.shields.io/github/workflow/status/drashland/sockets-client/master?label=ci">
+  </a>
 </p>
+
+---
+
+## Table of Contents
+- [Quick Start](#quick-start)
+- [Why use Sockets Client?](#why-use-sockets-client)
```
Comment (saragee3) [inline README.md]: again.. wocket, wocket?

**Comment by ebebbington** [comment]
@saragee3 ill change all the references later 👍

File: README.md (hunk 2 above)
Comment (ebebbington) [reply README.md]: https://github.com/drashland/sockets-client/pull/6/commits/1329969f0af29ca57ff4657e24a20f46809b8418

File: LICENSE (hunk 1 above)
Comment (ebebbington) [reply LICENSE]: https://github.com/drashland/sockets-client/pull/6/commits/1329969f0af29ca57ff4657e24a20f46809b8418

**Comment by ebebbington** [comment]
DONT REVIEW THIS. Making a PR to this PR to change the name to wocket-client

**Comment by ebebbington** [comment]
Turns out i already did, but left a few occurrences out.. ready for review again

File: LICENSE [hunk 3]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2019-2020 the sockets authors
+Copyright (c) 2020 the Wocket-client authors
```
Comment (crookse) [inline LICENSE]: 2020 Drash Land. see https://github.com/drashland/.github/blob/master/LICENSE

Comment (ebebbington) [reply LICENSE]: https://github.com/drashland/sockets-client/pull/6/commits/248781570fd97982b1cb2fc14fe19360167a2589

**Comment by ebebbington** [comment]
closing in favour of https://github.com/drashland/wocket/issues/111

**Pull request closed without merging**